tests/testsuite/login.rs

```diff
@@ -163,7 +163,7 @@ fn new_credentials_is_used_instead_old() {
         execs().with_status(0),
     );
 
-    let config = Config::new(Shell::new(), cargo_home(), cargo_home());
+    let config = Config::new(Shell::new(), CargoDirs::new().unwrap());
```

Comment:
Here I think we want to point `CargoDirs` to `CARGO_HOME` inside test root. 

One option is to provide another constructor for CargoDirs which accepts `cargo_home: PathBuf`. Another option is to modify the test such that it reads the config file directly. I am slightly in favor of the second approach.